resources/nspawn.go

```diff
@@ -0,0 +1,351 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	// "github.com/purpleidea/mgmt/global"
+	"github.com/purpleidea/mgmt/event"
+	"github.com/purpleidea/mgmt/util"
+
+	// systemd "github.com/coreos/go-systemd/dbus" // change namespace
+	systemdUtil "github.com/coreos/go-systemd/util"
+	"github.com/godbus/dbus" // namespace collides with systemd wrapper
+	machined "github.com/joejulian/go-systemd/machine1"
+)
+
+const (
+	running               = "running"
+	stopped               = "stopped"
+	dbusInterface         = "org.freedesktop.machine1.Manager"
+	dbusPath              = "/org/freedesktop/machine1"
+	machineNew            = "org.freedesktop.machine1.Manager.machineNew"
+	machineRemoved        = "org.freedesktop.machine1.Manager.machineRemoved"
+	nspawnServiceTemplate = "systemd-nspawn@%s"
+)
+
+func init() {
+	gob.Register(&NspawnRes{})
+}
+
+// NspawnRes is an nspawn container resource
+type NspawnRes struct {
+	BaseRes `yaml:",inline"`
+	State   string `yaml:"state"`
+}
+
+// Init runs some startup code for this resource.
+func (obj *NspawnRes) Init() error {
+	obj.BaseRes.kind = "Nspawn"
+	return obj.BaseRes.Init()
+}
+
+// NewNspawnRes is the constructor for this resource.
+func NewNspawnRes(name string, state string) (*NspawnRes, error) {
+	obj := &NspawnRes{
+		BaseRes: BaseRes{
+			Name: name,
+		},
+		State: state,
+	}
+	return obj, obj.Init()
+}
+
+// Validate params
+func (obj *NspawnRes) Validate() error {
+	switch obj.State {
+	case running, stopped:
+		break
+	default:
+		return fmt.Errorf("Invalid State: %s", obj.State)
+	}
+	return nil
+}
+
+// Watch for state changes and sends a message to the bus if there is a change.
+func (obj *NspawnRes) Watch(processChan chan event.Event) error {
+	// Boilerplate
+	// TODO: Boilerplate sucks. Find a way to prevent its necessity.
+	if obj.IsWatching() {
+		return nil
+	}
+	obj.SetWatching(true)
+	defer obj.SetWatching(false)
+	cuid := obj.converger.Register()
+	defer cuid.Unregister()
+
+	var startup bool
+	Startup := func(block bool) <-chan time.Time {
+		if block {
+			return nil // blocks forever
+		}
+		return time.After(time.Duration(500) * time.Millisecond) // 1/2 the resolution of converged timeout
+	}
+	// End Boilerplate
+
+	// This resource depends on systemd. Ensure that it's running.
+	// TODO: If systemd is not running, should we start it?
+	if !systemdUtil.IsRunningSystemd() {
+		return fmt.Errorf("Systemd is not running.")
+	}
+
+	// Create a private message bus
```

Comment:
removed capitalization and punctuation from all remarks this leaves them looking utterly unreadable it's just like trying to read the text messages of a teenager